bindings/matrix-sdk-ffi/src/client_login_builder.rs

```diff
@@ -0,0 +1,66 @@
+use std::sync::{Arc, RwLock};
+
+use matrix_sdk::matrix_auth::LoginBuilder;
+
+use crate::ClientError;
+
+/// Builder type used to configure optional settings for logging in to a matrix
+/// server
+#[derive(Clone, uniffi::Object)]
+pub struct ClientLoginBuilder {
+    /// Because [LoginBuilder] doesn't implement Clone the options where to
+    /// either make this a shadow version of LoginBuilder (with all the data
+    /// stored inline) or to store the builder in a way that makes it
+    /// consumable. By using [RwLock] with an [Option] inside the latter was
+    /// achieved with an error condition if [send] is called twice.
+    inner: Arc<RwLock<Option<LoginBuilder>>>,
```

Comment:
While this approach works, we've done it differently elsewhere in the FFI layer, and I'd like to keep it consistent with it. Can you cargo-cult what the `ClientBuilder` does, so we can avoid the lock, please?